<!doctype html>
<html lang="en">
  <head>
    <!-- Required meta tags -->
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.79.0">
    <title>ExplorePipolin</title>
    <link rel="canonical" href="https://getbootstrap.com/docs/5.0/examples/cheatsheet/">
    
    <!-- Bootstrap CSS -->
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-giJF6kkoqNQ00vy+HMDP7azOuL0xtbfIcaT9wjKHr8RbDVddVHyTfAAsrekwKmP1" crossorigin="anonymous">

    <style>
      .bd-placeholder-img {
        font-size: 1.125rem;
        text-anchor: middle;
        -webkit-user-select: none;
        -moz-user-select: none;
        user-select: none;
      }

      @media (min-width: 768px) {
        .bd-placeholder-img-lg {
          font-size: 3.5rem;
        }
      }
    </style>

    
    <!-- Custom styles for this template -->
    <link href="cheatsheet.css" rel="stylesheet">
  </head>
  <body class="bg-light">
    
<header class="bd-header bg-dark py-3 d-flex align-items-stretch border-bottom border-dark">
  <div class="container-fluid d-flex align-items-center justify-content-between">
    <p class="d-flex align-items-center fs-4 mb-0 ms-4" style="color:#bdd6fd">
      <strong>ExplorePipolin</strong>
    </p>
    <div>
      <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" fill="#bdd6fd" class="bi bi-github" viewBox="0 0 16 16">
      <path d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.012 8.012 0 0 0 16 8c0-4.42-3.58-8-8-8z"/>
      </svg>
       
      <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" fill="#bdd6fd" class="bi bi-twitter" viewBox="0 0 16 16">
        <path d="M5.026 15c6.038 0 9.341-5.003 9.341-9.334 0-.14 0-.282-.006-.422A6.685 6.685 0 0 0 16 3.542a6.658 6.658 0 0 1-1.889.518 3.301 3.301 0 0 0 1.447-1.817 6.533 6.533 0 0 1-2.087.793A3.286 3.286 0 0 0 7.875 6.03a9.325 9.325 0 0 1-6.767-3.429 3.289 3.289 0 0 0 1.018 4.382A3.323 3.323 0 0 1 .64 6.575v.045a3.288 3.288 0 0 0 2.632 3.218 3.203 3.203 0 0 1-.865.115 3.23 3.23 0 0 1-.614-.057 3.283 3.283 0 0 0 3.067 2.277A6.588 6.588 0 0 1 .78 13.58a6.32 6.32 0 0 1-.78-.045A9.344 9.344 0 0 0 5.026 15z"/>
      </svg>
      <img src="assets/rs_logo.svg" width="20" height="20" ></img>
    </div>
    <form class="d-flex">
            <input class="form-control me-2" type="search" placeholder="Search" aria-label="Search">
      <button class="btn btn-secondary" type="submit"><p class="mb-0" style="color:#c7d3e6">Search</p></button>
    </form>
  </div>
</header>
<aside class="bd-aside sticky-xl-top text-muted align-self-start mb-3 mb-xl-5 px-2">
  <h3 class="h6 pt-4 pb-3 mb-4 border-bottom" style="color:#485f88e8"><strong>List of contents</strong></h3>
  <nav class="small" id="toc">
    <ul class="list-unstyled">
      <li class="my-2">
        <button class="btn d-inline-flex align-items-center collapsed" data-bs-toggle="collapse" aria-expanded="false" data-bs-target="#introduction-collapse" aria-controls="introduction-collapse">Introduction</button>
        <ul class="list-unstyled ps-3 collapse" id="introduction-collapse">
          <li><a class="d-inline-flex align-items-center rounded" href="#welcome">Welcome to ExplorePipolin</a></li>
          <li><a class="d-inline-flex align-items-center rounded" href="#overview">Overview</a></li>
        </ul>
      </li>
      <li class="my-2">
        <button class="btn d-inline-flex align-items-center collapsed" data-bs-toggle="collapse" aria-expanded="false" data-bs-target="#installation-collapse" aria-controls="installation-collapse">Installation</button>
        <ul class="list-unstyled ps-3 collapse" id="installation-collapse">
          <li><a class="d-inline-flex align-items-center rounded" href="#requirements">Requirements</a></li>
          <li><a class="d-inline-flex align-items-center rounded" href="#installation-source">Installation from source</a></li>
          <li><a class="d-inline-flex align-items-center rounded" href="#installation-conda">Installation using Conda</a></li>
        </ul>
      </li>
      <li class="my-2">
        <button class="btn d-inline-flex align-items-center collapsed" data-bs-toggle="collapse" aria-expanded="false" data-bs-target="#usage-collapse" aria-controls="usage-collapse">Usage</button>
        <ul class="list-unstyled ps-3 collapse" id="usage-collapse">
          <li><a class="d-inline-flex align-items-center rounded" href="#input-files">Input files</a></li>
          <li><a class="d-inline-flex align-items-center rounded" href="#running-ExplorePipolin">Running ExplorePipolin</a></li>
          <li><a class="d-inline-flex align-items-center rounded" href="#test-run">Test run</a></li>
          <li><a class="d-inline-flex align-items-center rounded" href="#running-docker">Running with Docker</a></li>
        </ul>
      </li>
      <li class="my-2">
        <button class="btn d-inline-flex align-items-center collapsed" data-bs-toggle="collapse" aria-expanded="false" data-bs-target="#output-files-collapse" aria-controls="output-files-collapse">Output files</button>
        <ul class="list-unstyled ps-3 collapse" id="output-files-collapse">
          <li><a class="d-inline-flex align-items-center rounded" href="#output-files-summary">Summary</a></li>
          <li><a class="d-inline-flex align-items-center rounded" href="#output-files-atts">atts</a></li>
          <li><a class="d-inline-flex align-items-center rounded" href="#output-files-atts_denovo">atts_denovo</a></li>
          <li><a class="d-inline-flex align-items-center rounded" href="#output-files-logs">logs</a></li>
          <li><a class="d-inline-flex align-items-center rounded" href="#output-files-pipolbs">pipolbs</a></li>
          <li><a class="d-inline-flex align-items-center rounded" href="#output-files-pipolins">pipolins</a></li>
          <li><a class="d-inline-flex align-items-center rounded" href="#output-files-prokka">prokka</a></li>
          <li><a class="d-inline-flex align-items-center rounded" href="#output-files-trnas">trnas</a></li>
        </ul>
      </li>
      <li class="my-2">
        <button class="btn d-inline-flex align-items-center collapsed" data-bs-toggle="collapse" aria-expanded="false" data-bs-target="#examples-collapse" aria-controls="examples-collapse">Examples</button>
        <ul class="list-unstyled ps-3 collapse" id="examples-collapse">
          <li><a class="d-inline-flex align-items-center rounded" href="#one-genome-example">One genome example</a></li>
          <li><a class="d-inline-flex align-items-center rounded" href="#multiple-genomes-example">Multiple genomes example</a></li>
        </ul>
      </li>
      <li class="my-2">
        <button class="btn d-inline-flex align-items-center collapsed" data-bs-toggle="collapse" aria-expanded="false" data-bs-target="#FAQs-collapse" aria-controls="FAQs-collapse">FAQs</button>
        <ul class="list-unstyled ps-3 collapse" id="FAQs-collapse">
          <li><a class="d-inline-flex align-items-center rounded" href="#FAQs">FAQs</a></li>
        </ul>
      </li>
    </ul>
  </nav>
</aside>
<div class="bd-cheatsheet container-fluid bg-white">
  <section id="Introduction">
    <h5 class="sticky-xl-top fw-bold"></h5>
    <article class="my-3" id="welcome">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
        <h3 class="text-center" style="color:#0b2652">Welcome</h3>
      </div>
      
      <div>
        <div class="bd-example border border-3">
          <img src="assets/banner.svg" class="img-fluid"></img>
        </div>

        <div class="bd-example">
          <p class="lead text-center p-0">
            <strong>ExplorePipolin:</strong> a search tool for identifying and analyzing pipolins within bacterial genomes.
          </p>
        </div>
        <div class="row mt-0">
          <div class="col-sm-6 p-3">
            <div class="container rounded bg-light p-3 shadow">
              <p><strong>Download ExplorePipolin</strong></p>
              <div class="btn-group" role="group">
                <button id="btnGroupDrop1" type="button" class="btn btn-dark dropdown-toggle" data-bs-toggle="dropdown" aria-expanded="false">
                  Download
                </button>
                <ul class="dropdown-menu" aria-labelledby="btnGroupDrop1">
                  <li><a class="dropdown-item" href="#">.zip file</a></li>
                  <li><a class="dropdown-item" href="#">.tar.gz file</a></li>
                </ul>
              </div>
            </div>
          </div>
          <div class="col-sm-6 p-3">
            <div class="container rounded bg-light p-3 shadow">
              <p><strong>View project on GitHub</strong></p>
              <a href="#" class="btn btn-dark">GitHub</a>
            </div>
          </div>
        </div>
      </div>
      </article>

      <article class="my-3" id="overview">
         <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
           <h3 class="text-center" style="color:#0b2652">Overview</h3>
         </div> 

        <div>

          <div class="bd-example">
          <div class="accordion" id="accordionExample">
            <div class="accordion-item">
              <h4 class="accordion-header" id="headingOne">
                <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#collapseOne" aria-expanded="true" aria-controls="collapseOne">
                  <strong>What are Pipolins?</strong>
                </button>
              </h4>
              <div id="collapseOne" class="accordion-collapse collapse show" aria-labelledby="headingOne" data-bs-parent="#accordionExample">
                <div class="accordion-body">
                  Pipolins constitute a recently reported new group of integrative MGEs widespread among diverse bacterial
                  phyla and also identifed in mitochondria as circular plasmids. The hallmark feature of the pipolins is a gene
                  encoding for a replicative family B DNA polymerases (PolB) with an intrinsic de novo primer synthesis capacity
                  called primer-independent PolBs (piPolBs), hence their name (piPolB-encoding elements). Preliminary phylogenetic 
                  analyses indicated that piPolBs would form a third major branch of PolB, besides the protein-primed
                  PolBs (pPolBs), found in a wide range of viruses and plasmids, and RNA-primed PolBs (rPolBs), which are the
                  principal replicative enzymes of archaea and eukaryotes, some archaea and many dsDNA viruses.
                </div>
              </div>
            </div>
          </div>
          </div>
        
          <div class="list-group">
            <p class="list-group-item list-group-item-action list-group-item-primary p-3"><strong>Pipeline description</strong></p>
            <p>ExplorePipolin pipeline is intended to check automatically whether a set of bacterial genome assemblies contains pipolins.
               Then, it extracts and annotates the pipolins homogeneously. 
               To this end, the pipeline performs the following tasks:</p>
            <ol>
              <strong><li>Pipolin detection</li></strong>
              <p>Identification of main hallmarks:</p>
              <ol class="mt-2">
                <li>Prediction of the piPolB gene by BLAST search using a reference sequence.</li>
                <li>Att terminal direct repeats detection by BLAST using a reference Att. 
                  De novo search of att sites is performed if the previous search fails.</li>
                <li>Find tRNA genes with ARAGORN</li>
              </ol>
              <strong><li class="mt-4">Pipolin scaffolding</li></strong>
              <p>In some cases, att-repeats and piPolB were located on different contigs, 
                posing a challenge for us to understand the order and orientation of the contigs.
                In this step, we assume that att-repeats should be headed in the same direction as
                they are direct repeats and that one of them could overlap with a tRNA gene on the 
                opposite strand. For consistency, we referred to the latter att-repeat as attR and 
                expected it always to be the rightmost att.</p>
              <strong><li>Pipolin sequence extraction</li></strong>
              <p>After identification of hallmark features and scaffolding (if needed) the sequence is extracted</p>
              <strong><li>Pipolin annotation</li></strong>
              <p>Pipolin genes need to be thoroughly and homogeneously annotated. 
                Prokka pipeline fulfils this task annotating many prokaryotic genome features 
                (protein-coding genes, tRNAs, rRNAs and non-coding RNAs among others).</p>
              <strong><li>Att repeats annotation</li></strong>
              <p>The information about found att repeats need to be included in the annotation separately,
                 as gene prediction and annotation tools usually do not handle any repeats.</p>
              <strong><li>Visualizing with EasyFig</li></strong>
              <p>EasyFig provides comparative representation of the genetic structure of pipolins. 
                We created a colouring scheme for the common pipolin genes and other meaningful features 
                (att repeats and assembly gaps).</p>
            </ol>
          
            <div class="alert alert-warning alert-dismissible fade show" role="alert">
            Note: The current version (v.????) is designed and tested for E. coli pipolins, 
            but the flexibility of the current pipeline allows an easy modification to extend the pipeline
            for a wider range of bacteria in the future.
            <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
            </div>
          </div>

          <div class="card mt-2">
            <div class="card-header">
            <strong>References</strong>
            </div>
            <div class="card-body">
              <ul>
                <li><blockquote class="blockquote">
                <p class="small">Primer-independent DNA synthesis by a family B DNA polymerase from self-replicating Mobile genetic elements.
                </p>
                <footer class="blockquote-footer">Redrejo-Rodríguez, M., et al. in <cite title="Source Title"><a href="https://www.cell.com/cell-reports/fulltext/S2211-1247(17)31486-9">Cell reports, 2017.</a></cite></footer>
              </blockquote></li>
              <li><blockquote class="blockquote">
                <p class="small">High diversity and variability of pipolins among a wide range of pathogenic <i>Escherichia coli</i> strains.</p>
                <footer class="blockquote-footer">Flament-Simon, S.C., de Toro, M., Chuprikova, L., et al. in <cite title="Source Title"><a href="https://www.nature.com/articles/s41598-020-69356-6">Scientific Reports, 2020.</a></cite></footer>
              </blockquote></li>
              </ul>
            </div>
          </div>   
        </div>
      </article>
  </section>
  <section id="installation-and-requirements">
    <h2 class="sticky-xl-top fw-bold pt-3 pt-xl-5 pb-2 pb-xl-3 text-center" style="color:#0b2652">Installation</h2>
    
    <article class="my-3" id="requirements">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
        <h3 class="text-center" style="color:#0b2652">Requirements</h3>
      </div>
      
      <div>
        <div class="bd-example">
          <p>List of requiered dependencies:</p>

          <table class="table">
            <thead>
              <tr class="table-primary" style="color:#0b2652">
                <th scope="col">Program</th>
                <th scope="col">Version</th>
                <th scope="col">Installation</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th scope="row">pip</th>
                <td>(find version)</td>
                <td><code style="color:#000000">sudo apt install python3-pip</code></td>
              </tr>
              <tr>
                <th scope="row">HMMER</th>
                <td>"</td>
                <td><code>sudo apt install hmmer</code></td>
              </tr>
              <tr>
                <th scope="row">BLAST +</th>
                <td>"</td>
                <td><code>sudo apt install ncbi-blast+</code></td>
              </tr>
              <tr>
                <th scope="row">ARAGORN</th>
                <td>"</td>
                <td><code>sudo apt install aragorn</code></td>
              </tr>
              <tr>
                <th scope="row">Prokka</th>
                <td>"</td>
                <td><a href="https://github.com/tseemann/prokka">Prokka installation web page</a></td>
              </tr>
              <tr>
                <th scope="row">Prodigal</th>
                <td>"</td>
                <td><code>sudo apt install prodigal</code></td>
              </tr>
              <tr>
                <th scope="row">EasyFig (Optional)</th>
                <td>"</td>
                <td><a href="https://mjsull.github.io/Easyfig/">EasyFig installation web page</td>
              </tr>
            </tbody>
          </table>

        </div>
      
    </article>
  
    <article class="my-3" id="installation-source">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
        <h3 class="text-center" style="color:#0b2652">Installation from source</h3>
      </div>
       
      <div>
        <div class="bd-example">
        <ol>
          <li>Install the requirements (see above).</li>
          <li>Download software
          <ul><li><pre><code>wget https://github.com/liubovch/ExplorePipolin/archive/0.0.a1.zip.</pre></code></li></ul></li>
          <li>Unzip content.
          <ul><li><pre><code>unzip 0.0.a1.zip && cd ExplorePipolin-0.0.a1</pre></code></li></ul></li>
          <li>Installing.
          <ul>
            <li>Install in user site-package: <code>pip install .</code></li>
            <li> Requires superuser privileges: <code>sudo pip install .</code></li>
          </ul></li>
         </ol>
          <p>NOTE: before installing, it is possible to run unit tests:</p>
        <ul>
            <li><code>pytest</code></li>
            <li><code>python setup.py test</code> (from the source root directory)</li>
        </ul>
        <p>How to uninstall:
          <code>pip uninstall ExplorePipolin</code></p>
        </div>
       </div>
    </article>
      
    <article class="my-3" id="installation-conda">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
        <h3 class="text-center" style="color:#0b2652">Installation unsing Conda</h3>
      </div>
      
      <div>
        <div class="bd-example">
        <p>Before installing ExplorePipolin, make sure you are running the latest version of Conda</p>
        <ul>
            <li><code>conda update conda</code></li>
            <li><code>conda install wget</code></li>
        </ul>
      </div>
       
      <div>
        <div class="bd-example">
        <ol>
          <li>Create a new environment thar is specific for ExplorePipolin. You can choose whatever name you'd like for the environment.
            <ul>
              <li><code>wget https://github.com/liubovch/ExplorePipolin/releases/download/0.0.a1/explore-pipolin-0.0.a1-py_0.yml</code></li>
              <li><code>conda env create -n ExplorePipolin-0.0.a1 --file explore-pipolin-0.0.a1-py_0.yml</code></li>
            </ul></li>
         <li>Download and install ExplorePipolin into the created environment:
            <ul>
              <li><code>wget https://github.com/liubovch/ExplorePipolin/releases/download/0.0.a1/explore-pipolin-0.0.a1-py_0.tar.bz2</code></li>
              <li><code>conda install -n ExplorePipolin-0.0.a1 explore-pipolin-0.0.a1-py_0.tar.bz2</code></li>
            </ul></li>
         <li>Clean up (optional):
           <ul><li><code>rm explore-pipolin-0.0.a1-py_0.yml explore-pipolin-0.0.a1-py_0.tar.bz2</code></li></ul></li>
         <li>Activate the environment and check the installation:
           <ul>
             <li><code>conda activate ExplorePipolin-0.0.a1</code></li>
             <li><code>explore_pipolin -h</code></li>
           </ul></li>
          </div>
        </div>
    
   </section>
  <section id="usage">
    <h2 class="sticky-xl-top fw-bold pt-3 pt-xl-5 pb-2 pb-xl-3 text-center" style="color:#0b2652">Usage</h2>
    
    <article class="my-3" id="input-files">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2 text-center" style="color:#0b2652">
        <h3>Input files</h3>
      </div>

      <div class="bd-example">
        <p>ExplorePipolin can take one or multiple genome files in FASTA format.
          A genome sequence can be either a single complete chromosome (preferred) or contigs (in a single multiFASTA file). 
          Draft genome files are also accepted since our pipeline includes a scaffolding task.</p>
      </div>
    </article>

    <article class="my-3" id="running-explore-pipolin">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2 text-center" style="color:#0b2652">
        <h3>Running ExplorePipolin</h3>
      </div>

      <div class="bd-examle">
        <p>To use Explore Pipolin just run the program in the command line along with the genome files and the desired options:</p>
        <pre class="mt-0 bg-light"><code>
        <strong>~$ explore_pipolin [OPTIONS] GENOMES...</strong>
        </code></pre>
      </div>
    </article>


    <article class="my-3" id="test-run">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2 text-center" style="color:#0b2652">
        <h3>Test run</h3>
      </div>
      
      <div class="bd-example">
      <p>ExplorePipolin includes a test run explaining the usage and the current options available:</p>
      <pre class="mt-0 bg-light"><code>
        <strong>~$ explore_pipolin -h</strong> 
        Usage: explore_pipolin [OPTIONS] GENOME...

        ExplorePipolin is a search tool for prediction and analysis of pipolins,
        bacterial mobile genetic elements.
      
        Options:
          --out-dir PATH              Output directory  [required]
          --add-colours               Add colours to the final Genbank file features.
                                      The genomic structure can be visualized further
                                      using Easyfig.
          --pipolb-hmm-profile PATH   If not provided, the default profile will be
                                      used instead.
          --ref-att PATH              Att sequence in FASTA file to use as 1st
                                      priority. If not provided, the default file
                                      will be used instead.
          --percent-identity INTEGER  Minimum percent identity in direct repeats
                                      search  [default: 85]
          --proteins PATH             Prokka param: FASTA or GBK file to use as 1st
                                      priority. If not provided, the default file
                                      will be used instead.
          --cpus INTEGER              Prokka param: Number of CPUs to use [0=all]
                                      [default: 8]
          -h, --help                  Show this message and exit.
      
      </pre></code>
      </div>
    </article>
    
    <article class="my-3" id="running-docker">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
        <h3 class="text-center" style="color:#0b2652">Running with Docker</h3>
      </div>
      
      <div class="bd-example">
      <p>See <a href="https://docs.docker.com/install/">https://docs.docker.com/install/</a> to install Docker.</p>
      <p>NOTE: superuser privileges are required to run the analysis and around 3GB of disk space for the image.</p>
      <div class="bd-example bg-light"><pre><code>
      sudo docker pull docker.pkg.github.com/liubovch/explorepipolin/explore_pipolin:0.0.a1
      sudo docker tag docker.pkg.github.com/liubovch/explorepipolin/explore_pipolin:0.0.a1 explore_pipolin 
      sudo docker run --rm explore_pipolin -h
      sudo docker run --rm -v $(pwd):/output -w /output explore_pipolin 
      --out-dir output ./input_genomes/*.fa   #(example run)
      </code></pre></div>
      </div>
    </article>
      
  </section>

  <section id="output-files">
    <h2 class="sticky-xl-top fw-bold pt-3 pt-xl-5 pb-2 pb-xl-3 text-center" style="color:#0b2652">Output files</h2>
    
    <article class="my-3" id="output-files-summary">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
        <h3 class="text-center" style="color:#0b2652">Summary</h3>
      </div>

      <div class="db-example">
      <div class="mt-2">The output directory will contain several folders:</div>
        <div class="mt-2">
      <table class="table table-striped">
          <thead>
          <tr>
            <th scope="col">Folder</th>
            <th scope="col">Content description</th>
          </tr>
          </thead>
          <tbody>
          <tr>
            <th scope="row"><code style="color:#1e69af">atts</code></th>
            <td>BLAST search results for the known att sites</td>
          </tr>
          <tr>
            <th scope="row"><code style="color:#1e69af">atts_denovo</class></th>
            <td>Results of de novo search for att sites</td>
          </tr>
          <tr>
            <th scope="row"><code style="color:#1e69af">logs</class></th>
            <td>log file from ExplorePipolin run</td>
          </tr>
          <tr>
            <th scope="row"><code style="color:#1e69af">pipolbs</code></th>
            <td>BLAST search results for piPolB genes</td>
          </tr>
          <tr>
            <th scope="row"><code style="color:#1e69af">pipolin</class></th>
            <td>extracted pipolin sequences in FASTA format</td>
          </tr>
          <tr>
            <th scope="row"><code style="color:#1e69af">prokka</class></th>
            <td>Prokka annotation results (check files description <a href="https://github.com/tseemann/prokka/blob/master/README.md#output-files">here</a>)</td>
          </tr>
          <tr>
            <th scope="row"><code style="color:#1e69af">trnas</class></th>
            <td>ARAGORN search results for tRNAs/tmRNAs</td>
          </tr>
          </tbody>
        </table>
      </div>
      </div>

    </article>

    
    <article class="my-3" id="output-files-atts">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
        <h3 class="text-center" style="color:#0b2652">atts</h3>
      </div>

      <div class="bd-example mt-2">
      <p>The <strong style="color:#1e69af">atts</strong> folder contains all the results from the BLAST att search. 
      These are known att sites, similar to the one given as reference in the running command 
      (or to the <i>E. coli</i> 3-373-03-S1_C2 pipolin by default).
      Inside this folder we can find one BLAST output file for each genome given in the input.</p>
      <p>The results are presented in BLAST output format 5 (XML). Here we show part of an example from the 
        <i>E. coli</i> 3-373-03-S1_C2 atts:
        <div class="bd-example mt-0 border border-2">
          <img src="images/Ec_3-373-fmt5.png" class="img-fluid"></img>
        </div>
      </p>
      </div>
    </article>

    <article class="my-3" id="output-files-atts_denovo">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
        <h3 class="text-center" style="color:#0b2652">atts_denovo</h3>
      </div>

      <div class="bd-example mt-2">
      <p>The <strong style="color:#1e69af">atts_denovo</strong> folder contains the results from the <i>de novo</i> att search. 
      Inside this folder we can find 5 files for each genome given in the input. Again, we've presented here part of each file
      from the <i>E. coli</i> 3-373-03-S1_C2 example:
      <ul class="mt-0">
        <li>.atts_denovo file (CAMBIAR)</li>
        <div class="bd-example mt-2 border border-2">
          <img src="images/Ec_3-373-fmt5.png" class="img-fluid"></img>
        </div>
      </ul>
      </p>
      </div>
    </article>

    <article class="my-3" id="output-files-logs">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
        <h3 class="text-center" style="color:#0b2652">logs</h3>
      </div>

      <div class="bd-example mt-2">
      <p>Inside the <strong style="color:#1e69af">logs</strong> folder you can find 
        one .log file for each genome input. This files containts all the information
        related to the task execution of ExplorePipolin (similar to the default 
        output of the program). 
      </p>
      <p class="mt-0">Example log file from the <i>E. coli</i> 3-373-03-S1_C2 run:</p>
      <div class="bd-example mt-2 border border-2">
        <img src="images/Ec_3-373-log.png" class="img-fluid"></img>
      </div>
      </div>
    </article>

    <article class="my-3" id="output-files-pipolbs">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
        <h3 class="text-center" style="color:#0b2652">pipolbs</h3>
      </div>

      <div class="bd-example mt-2">
      <p>The <strong style="color:#1e69af">pipolbs</strong> folder contains the search results
        fot piPolB genes performed by BLAST. Here we can find 2 files for each genome input:</p>
      <ul class="mt-0">
        <li>.faa files: multiFASTA files with all the BLAST hits. 
          <i>E. coli</i> 3-373-03-S1_C2 example:
        </li>
      </ul>
        <div class="bd-example mt-2 border border-2">
          <img src="images/Ec_3-373-pipolb-faa.png" class="img-fluid"></img>
        </div>
        <ul class="mt-4">
          <li>.tbl files: (Ask explanation)
            <i>E. coli</i> 3-373-03-S1_C2 example:
          </li>
        </ul>
          <div class="bd-example mt-2 border border-2">
            <img src="images/Ec_3-373-pipolb-tbl.png" class="img-fluid"></img>
          </div>
      </div>
    </article>

    <article class="my-3" id="output-files-pipolins">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
        <h3 class="text-center" style="color:#0b2652">pipolins</h3>
      </div>

      <div class="bd-example mt-2">
      <p>The <strong style="color:#1e69af">pipolins</strong> folder contains:
      </p>
      </div>
    </article>

    <article class="my-3" id="output-files-prokka">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
        <h3 class="text-center" style="color:#0b2652">prokka</h3>
      </div>

      <div class="bd-example mt-2">+
      <p>The <strong style="color:#1e69af">prokka</strong> folder contains:
      </p>
      </div>
    </article>

    <article class="my-3" id="output-files-trnas">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
        <h3 class="text-center" style="color:#0b2652">trnas</h3>
      </div>

      <div class="bd-example mt-2">
      <p>The <strong style="color:#1e69af">trnas</strong> folder contains:
      </p>
      </div>
    </article>

  </section>

  <section id="Examples">
    <h2 class="sticky-xl-top fw-bold pt-3 pt-xl-5 pb-2 pb-xl-3 text-center" style="color:#0b2652">Examples</h2>
    
    <article class="my-3" id="one-genome-example">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
        <h3 class="text-center" style="color:#0b2652">One genome</h3>
      </div>
      
      <div>
        <div class="bd-example">
          <p>Here we provide an example of running ExplorePipolin with the <i>Escherichia coli</i>
             3-373-03-S1_C2 <a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000700265.1/">genome</a>:</p>
          <ol>
            <li>Run ExplorePipolin</li>
        <pre class="mt-2 bg-light"><code>
        <strong>~$ explore_pipolin E_coli.fa --out-dir Results --percent-identity 100 --add-colours</strong>
        </pre></code>
            <li>Tasks processing</li>
        <pre class="mt-2 bg-light"><code>
          [2021-02-23 12:22:25+0100] INFO - prefect.FlowRunner | Beginning Flow run for 'MAIN'
          [2021-02-23 12:22:25+0100] INFO - prefect.TaskRunner | Task 'pipolb_hmm_profile':
             Starting task run...
          [2021-02-23 12:22:26+0100] INFO - prefect.TaskRunner | Task 'pipolb_hmm_profile': 
             Finished task run for task with final state: 'Success'
          [2021-02-23 12:22:26+0100] INFO - prefect.TaskRunner | Task 'cpus': 
             Starting task run...
          [2021-02-23 12:22:26+0100] INFO - prefect.TaskRunner | Task 'cpus': 
             Finished task run for task with final state: 'Success'
          [2021-02-23 12:22:26+0100] INFO - prefect.TaskRunner | Task 'out_dir': 
             Starting task run...
          [2021-02-23 12:22:26+0100] INFO - prefect.TaskRunner | Task 'out_dir': 
             Finished task run for task with final state: 'Success'
          ...
        </pre></code>
        <p>See <a href="example_files/output.log">output.log</a> for the complete running output.</p>
            <li>Analyze results</li>
        <pre class="mt-2 bg-light"><code>
        <strong>~$ ls Results</code>
        <code style="color:#1e69af">atts  atts_denovo  logs  pipolbs  pipolins  prokka  trnas     
        </code></strong></pre>
        <p>See "Output files" section for more information.</p>
        <li>Visualize pipolin genetic structure with EasyFig (legend added manually):</li>
        <div class="bd-example mt-2 border border-2">
          <img src="images/Ec_3-373_EasyFig.png" class="img-fluid"></img>
        </div>
          </ol>
        </div>
      </div>

    </article>

    <article class="my-3" id="multiple-genomes-example">
      <div class="bd-heading sticky-xl-top align-self-start mt-5 mb-3 mt-xl-0 mb-xl-2">
        <h3 class="text-center" style="color:#0b2652">Multiple genomes</h3>
      </div>

      <div>
        <div class="bd-example">
          <p>We also wanted to show an example of running ExplorePipolin with multiple genomes.
            To this purpose we have selected the genomes from the following <i>E. coli</i> strains:
          </p>
          <ul>
            <li><i>Escherichia coli</i> <a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000700265.1/">3-373-03-S1_C2</a></li>
            <li><i>Escherichia coli</i> <a href="https://www.ncbi.nlm.nih.gov/nuccore/CP024256.1/">ATCC 43886</a></li>
            <li><i>Escherichia coli</i> <a href="https://www.ncbi.nlm.nih.gov/nuccore/CP029687.1/">E706</a></li>
            <li><i>Escherichia coli</i> <a href="https://www.ncbi.nlm.nih.gov/nuccore/CP034745.1/">L100</a></li>
            <li><i>Escherichia coli</i> <a href="https://www.ncbi.nlm.nih.gov/nuccore/CP002291.1/">P12b</a></li>
          </ul>
          <ol>
            <li>Run ExplorePipolin</li>
        <pre class="mt-2 bg-light"><code>
        <strong>~$ explore_pipolin *.fa --out-dir Results_multiple_genomes --percent-identity 100 --add-colours</strong>
        </pre></code>
            <li>Tasks processing</li>
        <pre class="mt-2 bg-light"><code>
          [2021-02-24 12:52:29+0100] INFO - prefect.FlowRunner | Beginning Flow run for 'MAIN'
          [2021-02-24 12:52:29+0100] INFO - prefect.TaskRunner | Task 'percent_identity': 
             Starting task run...
          [2021-02-24 12:52:29+0100] INFO - prefect.TaskRunner | Task 'percent_identity': 
             Finished task run for task with final state: 'Success'
          [2021-02-24 12:52:29+0100] INFO - prefect.TaskRunner | Task 'add_colours': 
             Starting task run...
          [2021-02-24 12:52:29+0100] INFO - prefect.TaskRunner | Task 'add_colours': 
             Finished task run for task with final state: 'Success'
          [2021-02-24 12:52:29+0100] INFO - prefect.TaskRunner | Task 'genome_file': 
             Starting task run...
          [2021-02-24 12:52:29+0100] INFO - prefect.TaskRunner | Task 'genome_file': 
             Finished task run for task with final state: 'Success'
          [2021-02-24 12:52:29+0100] INFO - prefect.TaskRunner | Task 'ref_att': 
             Starting task run...
          [2021-02-24 12:52:29+0100] INFO - prefect.TaskRunner | Task 'ref_att': 
             Finished task run for task with final state: 'Success'
          [2021-02-24 12:52:29+0100] INFO - prefect.TaskRunner | Task 'out_dir': 
             Starting task run...
          [2021-02-24 12:52:29+0100] INFO - prefect.TaskRunner | Task 'out_dir': 
             Finished task run for task with final state: 'Success'
          [2021-02-24 12:52:29+0100] INFO - prefect.TaskRunner | Task 'cpus': 
             Starting task run...
          [2021-02-24 12:52:29+0100] INFO - prefect.TaskRunner | Task 'cpus': 
             Finished task run for task with final state: 'Success'
          [2021-02-24 12:52:30+0100] INFO - prefect.TaskRunner | Task 'create_genome': 
             Starting task run...
          [2021-02-24 12:52:30+0100] INFO - prefect.TaskRunner | Task 'create_genome': 
             Finished task run for task with final state: 'Mapped'
          [2021-02-24 12:52:30+0100] INFO - prefect.TaskRunner | Task 'create_genome[0]': 
             Starting task run...
          [2021-02-24 12:52:30+0100] INFO - prefect.TaskRunner | Task 'create_genome[0]': 
             Finished task run for task with final state: 'Success'
          [2021-02-24 12:52:30+0100] INFO - prefect.TaskRunner | Task 'create_genome[1]': 
             Starting task run...
          [2021-02-24 12:52:30+0100] INFO - prefect.TaskRunner | Task 'create_genome[1]': 
             Finished task run for task with final state: 'Success'
          [2021-02-24 12:52:30+0100] INFO - prefect.TaskRunner | Task 'create_genome[2]': 
             Starting task run...
          [2021-02-24 12:52:30+0100] INFO - prefect.TaskRunner | Task 'create_genome[2]': 
             Finished task run for task with final state: 'Success'
          [2021-02-24 12:52:30+0100] INFO - prefect.TaskRunner | Task 'create_genome[3]': 
             Starting task run...
          [2021-02-24 12:52:30+0100] INFO - prefect.TaskRunner | Task 'create_genome[3]': 
             Finished task run for task with final state: 'Success'
          [2021-02-24 12:52:30+0100] INFO - prefect.TaskRunner | Task 'create_genome[4]': 
             Starting task run...
          [2021-02-24 12:52:30+0100] INFO - prefect.TaskRunner | Task 'create_genome[4]': 
             Finished task run for task with final state: 'Success'
          ...
        </pre></code>
        <p>See <a href="example_files/output.log">output.log</a> for the complete running output.</p>
            <li>Analyze results</li>
        <pre class="mt-2 bg-light"><code>
        <strong>~$ ls Results</code>
        <code style="color:#1e69af">atts  atts_denovo  logs  pipolbs  pipolins  prokka  trnas     
        </code></strong></pre>
        <p>Again, see "Output files" section for more information.</p>
          </ol>
        </div>
      </div>

    </article>
  </section>

  <section id="Issues">
    <h2 class="sticky-xl-top fw-bold pt-3 pt-xl-5 pb-2 pb-xl-3 text-center" style="color:#0b2652">Issues</h2>

    <div class="bd-example mt-5">
      <div class="mt-2">We encourage you to report any problem related to the installing or running of ExplorePipolin
        to the issues <a href="https://github.com/liubovch/ExplorePipolin/issues">web page</a>
      </div>
    </div>

  <section id="FAQs">
    <h2 class="sticky-xl-top fw-bold pt-3 pt-xl-5 pb-2 pb-xl-3 text-center" style="color:#0b2652">FAQs</h2>
  </section>

      <section id="footer">
        <h2 class="sticky-xl-top fw-bold pt-3 pt-xl-5 pb-2 pb-xl-3"></h2>
        
        <article class="my-3 bg-light" id="footer-a">
        <p></p>

        <footer class="container py-5 bg-light">
          <div class="col-12 col-md">
            <!--logo-->
            <small class="d-block mb-3 text-muted">Usefull links</small>
          </div>
        <div class="row">
          <div class="col-6 col-md">
            <h5>Lista de cosas</h5>
            <ul class="list-unstyled text-small">
              <li><a class="link-secondary" href="#">Cool stuff</a></li>
              <li><a class="link-secondary" href="#">Random feature</a></li>
              <li><a class="link-secondary" href="#">Team feature</a></li>
              <li><a class="link-secondary" href="#">Stuff for developers</a></li>
              <li><a class="link-secondary" href="#">Another one</a></li>
              <li><a class="link-secondary" href="#">Last time</a></li>
            </ul>
          </div>
          <div class="col-6 col-md">
            <h5>Lista 2</h5>
            <ul class="list-unstyled text-small">
              <li><a class="link-secondary" href="#">Resource name</a></li>
              <li><a class="link-secondary" href="#">Resource</a></li>
              <li><a class="link-secondary" href="#">Another resource</a></li>
              <li><a class="link-secondary" href="#">Final resource</a></li>
            </ul>
          </div>
          <div class="col-6 col-md">
            <h5>Información adicional</h5>
            <p class="d-block mb-3 text-muted">Cosas finales</p>
          </div>
        </div>
        </footer>
        </article>
          </section>
      



</div>    

      <script src="./assets/dist/js/bootstrap.bundle.min.js"></script>
    
      <script src="cheatsheet.js"></script>
  
  </body>
</html>

